accountsdb-plugin-interface/src/accountsdb_plugin_interface.rs

```diff
@@ -0,0 +1,78 @@
+/// The interface for AccountsDb plugins. A plugin must implement
+/// the AccountsDbPlugin trait to work with the Solana Validator.
+/// In addition, the dynamic libraray must export a "C" function _create_plugin which
+/// creates the implementation of the plugin.
+use {std::any::Any, std::io, thiserror::Error};
+
+#[derive(Clone, PartialEq, Default, Debug)]
+pub struct ReplicaAccountMeta {
+    pub pubkey: String,
+    pub lamports: u64,
+    pub owner: String,
+    pub executable: bool,
+    pub rent_epoch: u64,
+}
+
+impl Eq for ReplicaAccountInfo {}
+
+#[derive(Clone, Default, PartialEq, Debug)]
+pub struct ReplicaAccountInfo {
+    pub account_meta: ReplicaAccountMeta,
+    pub data: Vec<u8>,
```

Comment:
I mean the answer is yes, we will be held to backwards compatibility where whether we want to or not

Reply:
Thanks Trent, will look into that. This interface is binary contract using direct function calls. Such changes need to be communicated and solved via re-compiling the code, and fix potential conflicts. Out of this consideration, I declare the interfaces directly instead of using various existing account structures.

Reply:
Changed to use enum to pass versioned account info